servo/connector.py

```diff
@@ -386,7 +386,10 @@ def _routes_for_connectors_descriptor(connectors) -> Dict[str, "BaseConnector"]:
             if _validate_class(connector):
                 connector_routes[connector.__default_name__] = connector
             elif connector_class := _connector_class_from_string(connector):
-                connector_routes[connector_class.__default_name__] = connector_class
+                try:
+                    connector_routes[connector_class.name.lower()] = connector_class
+                except AttributeError:
+                    connector_routes[connector_class.__default_name__] = connector_class
```

Comment:
I'm not sure if this worked for you in testing but there are a few issues with this update:

* This particular routes method `_routes_for_connectors_descriptor` is associated with the configuration of multiple instances of a single connector and is not used in most cases: https://github.com/opsani/servox/#running-multiple-connector-instances
    * In most cases, the routes will come from here https://github.com/opsani/servox/blob/796946547c97e513780b6a6175d2edbd9bd0252d/servo/connector.py#L489-L496
* Numerous unit tests are currently failing because this update throws off the expected routes of connectors with no metadata.name specified. Eg. [this test connector](https://github.com/opsani/servox/blob/796946547c97e513780b6a6175d2edbd9bd0252d/tests/servo_test.py#L31) now has a route of firsttestservo instead of first_test_servo
* In most cases, `__default_name__` is set by `_name_for_connector_class` which already uses the connector_class.name property: https://github.com/opsani/servox/blob/796946547c97e513780b6a6175d2edbd9bd0252d/servo/connector.py#L301-L302 The reason metadata's name is being ignored comes down to order of operations:
    1. In most cases, `__default_name__` is set during `BaseConnector.__init_subclass__`: https://github.com/opsani/servox/blob/796946547c97e513780b6a6175d2edbd9bd0252d/servo/connector.py#L189
    2. However, metadata is not processed until after `BaseConnector.__init_subclass__` runs so the name used by `_name_for_connector_class` is the default value for name from here: https://github.com/opsani/servox/blob/796946547c97e513780b6a6175d2edbd9bd0252d/servo/connector.py#L186
    3. Once the metadata decorator sets the name, there is no logic to update the `__default_name__` with the new name from metadata

Given the above points, I think this should be reverted and logic should be added to the [metadata decorator](https://github.com/opsani/servox/blob/796946547c97e513780b6a6175d2edbd9bd0252d/servo/connector.py#L248-L294) to update `__default_name__` like such:

```
        if name:
            if isinstance(name, tuple):
                if len(name) != 2:
                    raise ValueError(
                        f"Connector names given as tuples must contain exactly 2 elements: full name and alias"
                    )
                cls.name, cls.__default_name__ = name
            else:
                cls.name = name
            cls.__default_name__ = _name_for_connector_class(cls)
```